### ISSUE TITLE
"Heap corruption detected: pointer 0x7f00ba5ae400 to past fromspace

### ISSUE THREAD

**Issue by JJ** [MoarVM/MoarVM#822]
[This program](https://github.com/JJ/p6-algorithm-evolutionary-simple/blob/master/examples/concurrent-evolutionary-algorithm.p6) which is part of the Algorithm::Evolutionary::Simple package kind of works when it does. However, it fails from time to time with that error above, core dumps or different errors. Another one: 
`MoarVM panic: Internal error: zeroed target thread ID in work pass`
or simply
`[2]    10825 segmentation fault (core dumped)  perl6 -Ilib examples/concurrent-evolutionary-algorithm.p6`
Fails two out of three times, something like that.

**Comment by JJ** [comment]
This has been apparently fixed in 2018.03. Several runs, no problem. Great! :-)

**Comment by AlexDaniel** [comment]
Yeah I've noticed that too. The output of that script is not stable across runs though.

**Comment by JJ** [comment]
It's an evolutionary algorithm, it shouldn't be :-)

**Comment by AlexDaniel** [comment]
Yeah but in some cases it gets stuck? Let me try that again…

**Comment by JJ** [comment]
Yes, it does. Goes with the evolutionary algorithm. I mean, it gets stuck
in a place where it cannot go out, working with the same individuals over
and over again...
Come to think of it... I forgot mutation. Thanks!

**Comment by AlexDaniel** [comment]
OK, I guess this can be closed then. I'll leave it open for a bit with `testneeded` tag in case someone wants to try golfing it (or at least bisecting to see which commit did the trick).
